.github/workflows/ci.yaml

```diff
@@ -60,6 +62,12 @@ jobs:
             c_compiler: cl
     steps:
       - uses: actions/checkout@v4
+      - name: Install OpenMP on Ubuntu
```

Comment:
Something like

```diff
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a085d75..eb376f7 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -28,6 +28,11 @@ if(MATH_LIBRARY)
   target_link_libraries(topotoolbox PUBLIC ${MATH_LIBRARY})
 endif()
 
+find_package(OpenMP)
+if (OpenMP_C_FOUND)
+  target_link_libraries(topotoolbox PUBLIC OpenMP::OpenMP_C)
+endif()
+
 if (BUILD_SHARED_LIBS)
 else()
   target_compile_options(topotoolbox PUBLIC -DTOPOTOOLBOX_STATICLIB)
```

Seems to compile everything properly, except on macOS, where CMake struggles to find the Homebrew installed OpenMP.

Reply:
And adding this to the CI workflow gets it compiling on macOS:

```diff
diff --git a/.github/workflows/ci.yaml b/.github/workflows/ci.yaml
index 0a77b32..6219465 100644
--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -68,6 +68,9 @@ jobs:
       - name: Install OpenMP on macOS
         if: matrix.os == 'macos-latest'
         run: brew install libomp
+      - name: Set OpenMP_ROOT on macOS
+        if: matrix.os == 'macos-latest'
+        run: echo "OpenMP_ROOT=$(brew --prefix)/opt/libomp" >> $GITHUB_ENV
       - name: Install ninja on windows
         uses: seanmiddleditch/gha-setup-ninja@master
         if: matrix.os == 'windows-latest'
```

`OpenMP_ROOT` is a CMake environment variable that tells CMake where to look for the OpenMP library.

You also need to bump the minimum CMake version to avoid some warnings:

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2ed3f04..6c369ba 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.12)
 
 project(
   topotoolbox
```